### DISCUSSION TITLE
Why is captcha required on signin but disallowed when provided?

### DISCUSSION THREAD

**Discussion by Rc85** [supabase/supabase#20816] [Questions]
I have hcaptcha enabled and when I try to use the `signInWithPassword` method, it returns
```
code: 500
error_id: "84cf96e2655c7107-YVR"
msg: "captcha verification process failed"
```
The method doesn't allow options to be provided but when I do, I get
```
code: 400
msg: "captcha protection: request disallowed (invalid-input-secret)"
```
Is it because I'm sending request from localhost?

**Comment by santiarr** [comment]
bump

**Comment by lior-amsalem** [comment]
a good question, i have the same problem

i've confirm: secret & app keys
i've added localhost to cloudflare
i've used the dummy key for local host
(and am getting dummy token submitted to supabase)
i've made sure supabase key of cloudflare is correct.

**Comment by NinjaLukeI** [comment]
bump

**Comment by sanqiushili** [comment]
bump

**Comment by Ercilan** [comment]
`invalid-input-secret`

You need to make sure your supabase/config.toml is configured correctly:

```yaml
# Configure one of the supported captcha providers: `hcaptcha`, `turnstile`.
[auth.captcha]
enabled = true
provider = "turnstile"
# the turnstile testing screct
secret = "1x0000000000000000000000000000000AA"
```

The value of `auth.captcha.secret` (or the secret set in the web dashboard) must match the public key you are using on the frontend.
